File: 20221026/모범답안/20221026_05/piano.js

/**
 * 
 * @param s {string}
 * @return {string} 
 */
function solution(s) {
    let groupCount = 0;
    let currentGroupStart = -3;
    
    for (let i = 0; i < s.length; i++) {
        if (s[i] === '0') {
            continue;
        }

        if (currentGroupStart + 2 < i) {
            groupCount += 1;
            currentGroupStart = i;
        }
    }

    return groupCount <= 2 ? 'YES' : 'NO';
}

File: 20221026/모범답안/20221026_05/piano.py
def solution(s):
    '''
    :param s: string
    :return: string
    '''

    group_count = 0
    current_group_start_index = -3

    for idx, c in enumerate(s):
        if c == '0':
            continue
        
        if current_group_start_index + 2 < idx:
            group_count += 1
            current_group_start_index = idx
    
    return 'YES' if group_count <= 2 else 'NO'